File: 2026POTD/JANUARY/03-01-2026/1411.js
/**
 * @param {number} n
 * @return {number}
 */
var numOfWays = function(n) {
    const MOD = 1000000007;
    let aba = 6, abc = 6;
    for (let i = 2; i <= n; i++) {
        const newAba = (aba * 3 + abc * 2) % MOD;
        const newAbc = (aba * 2 + abc * 2) % MOD;
        aba = newAba;
        abc = newAbc;
    }
    return (aba + abc) % MOD;
};

File: 2026POTD/JANUARY/03-01-2026/1411.go
func numOfWays(n int) int {
    const MOD int64 = 1000000007
    var aba int64 = 6
    var abc int64 = 6
    for i := 2; i <= n; i++ {
        newAba := (aba*3 + abc*2) % MOD
        newAbc := (aba*2 + abc*2) % MOD
        aba = newAba
        abc = newAbc
    }
    return int((aba + abc) % MOD)
}